### PR TITLE
samples: matter: Created configuration with LEDs disabled

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Added Kconfig that allows to disable LEDs initialization and created a configuration that uses it for nRF54L15. This is required for the power measurement purposes.

### CODE REVIEW

File: doc/nrf/protocols/matter/getting_started/low_power_configuration.rst [hunk 1]
```diff
@@ -202,6 +202,17 @@ Disable unused pins and peripherals
    :start-after: disable_unused_pins_start
    :end-before: disable_unused_pins_end
 
+Disable LEDs module
+===================
+
+When performing the power measurements on variouis development kits, the LEDs can be included in the measurement circuit or not:
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:208]:
```suggestion
When performing power measurements on various development kits, the LEDs can either be included in the measurement circuit or not:
```

File: doc/nrf/protocols/matter/getting_started/low_power_configuration.rst [hunk 2]
```diff
@@ -202,6 +202,17 @@ Disable unused pins and peripherals
...
+
+* For the nRF52840 DK and nRF5340 DK the LEDs are excluded from the measurement circuit, so they can be enabled for the low power configuration and it is not going to impact the measurement results.
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:210]:
```suggestion
* For the nRF52840 DK and nRF5340 DK, the LEDs are excluded from the measurement circuit, so they can be enabled for low power configuration without impacting the measurement results.
```

File: doc/nrf/protocols/matter/getting_started/low_power_configuration.rst [hunk 3]
```diff
@@ -202,6 +202,17 @@ Disable unused pins and peripherals
...
+* For the nRF54L15 DK, the MOSFET transistors controlling the LEDs are included in the measurement circuit, what results in increased measurement result by an additional, small leakage current, appearing if the LED is turned on.
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:211]:
```suggestion
* For the nRF54L15 DK, the MOSFET transistors controlling the LEDs are included in the measurement circuit, which results in increased measurement results due to a small leakage current if the LED is turned on.
```

File: doc/nrf/protocols/matter/getting_started/low_power_configuration.rst [hunk 4]
```diff
@@ -202,6 +202,17 @@ Disable unused pins and peripherals
...
+  To measure the current consumption of the nRF54L15 SoC and not include development kit components, such as LEDs, it is recommended to disable them.
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:212]:
```suggestion
  To measure the current consumption of the nRF54L15 SoC without including development kit components, such as LEDs, it is recommended to disable them.
```

File: doc/nrf/protocols/matter/getting_started/low_power_configuration.rst [hunk 5]
```diff
@@ -202,6 +202,17 @@ Disable unused pins and peripherals
...
+
+To disable LEDs from the Matter samples and applications, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_LEDS` Kconfig option to ``n``.
```
Comment (ArekBalysNordic) [inline doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:214]:
```suggestion
To disable LEDs in the Matter samples and applications, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_LEDS` Kconfig option to ``n``.
```

**Review by ArekBalysNordic — APPROVED** [review]
Just documentation suggestions, so approving.

**Review by maciejbaczmanski — APPROVED** [review]

File: doc/nrf/protocols/matter/getting_started/low_power_configuration.rst [hunk 6]
```diff
@@ -202,6 +202,17 @@ Disable unused pins and peripherals
    :start-after: disable_unused_pins_start
    :end-before: disable_unused_pins_end
 
+Disable LEDs module
+===================
+
+When performing the power measurements on various development kits, the LEDs can either be included in the measurement circuit or not:
+
+* For the nRF52840 DK and nRF5340 DK, the LEDs are excluded from the measurement circuit, so they can be enabled for the low power configuration and it is not going to impact the measurement results.
+* For the nRF54L15 DK, the MOSFET transistors controlling the LEDs are included in the measurement circuit, what results in increased measurement result by an additional, small leakage current, appearing if the LED is turned on.
```
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:211]:
```suggestion
* For the nRF54L15 DK, the MOSFET transistors controlling the LEDs are included in the measurement circuit.
  This results in measurement results being increased by an additional, small leakage current that appears if an LED is turned on.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]
Could also add a changelog entry noting that the Kconfig option was added to all Matter samples.

**Comment by kkasperczyk-no** [comment]
> Could also add a changelog entry noting that the Kconfig option was added to all Matter samples.

Sure, applied.

**Review by wiba-nordic — APPROVED** [review]

**Review by doublemis1 — APPROVED** [review]

**Review by umapraseeda — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**